core/vm/evm.go

```diff
@@ -450,6 +452,12 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if evm.StateDB.GetNonce(address) != 0 || (contractHash != (common.Hash{}) && contractHash != emptyCodeHash) {
 		return nil, common.Address{}, 0, ErrContractAddressCollision
 	}
+	// Check that [caller] has the right to deploy contracts
+	modifyStatus := getAllowListStatus(evm.StateDB, caller.Address())
+	if !modifyStatus.HasDeployPrivileges() {
```

Comment:
I think we only need to check this if there are any admins? Otherwise, this prevents the use of `subnet-evm` for chains with permissionless deploys.

Reply:
Yup, updating this to be based off of `IsAllowListEnabled`.